plotly/tools.py

```diff
@@ -1000,8 +1000,7 @@ def _get_anchors(r, c, x_cnt, y_cnt, shared_xaxes, shared_yaxes):
     # Function pasting x/y domains in layout object (2d case)
     def _add_domain(layout, x_or_y, label, domain, anchor, position):
         name = label[0] + 'axis' + label[1:]
-        graph_obj = '{X_or_Y}Axis'.format(X_or_Y=x_or_y.upper())
-        axis = getattr(graph_objs, graph_obj)(domain=domain)
+        axis = {'domain': domain}
```

Comment:
this isn't about memoization as much as it's about performance. the line `graph_obj = '{X_or_Y}Axis'.format(X_or_Y=x_or_y.upper())` was eating up a ton of cpu time because it was instantiating that `graph_object` and then immediately setting it inside `layout`, which effectively was instantiating it again.